Name Visibility + Create / Destroy detection

WarperSan opened the pull request:
# New Features
## Name Visibility 
In my testing, I've found it is quite annoying to have the name always displayed. It breaks the immersion if, for example, the loader is hidden in a structure. To solve this, I've added the rule `always_show_loader_name`, which determines if the custom name is always displayed or not.

This rule simply overrides the `setCustomNameVisible()` value, which was always `true`.

*Note: the rule doesn't change the visibility of existing loaders.*

## Create / Destroy detection
In my testing, I've found the mod doesn't have a way to detect when a loader is created nor destroyed. It would be impossible to reliably detect if a cart is a loader. 

To fix this, the dispenser now sends vibrations depending on the action. Right now, creating a loader uses the frequency `6`, and destroying a loader uses the frequency `5`.

It is possible to toggle this feature using the rule `emit_vibration_on_loader_change`. If set to `false`, no vibration will be emitted when starting or stopping a loader.

*Note: both actions can be triggered at once (if you create and destroy with the same tick). However, Doing the same action multiple times in the same tick will only trigger the vibration once.*

*Note 2: the rule controls both vibrations. It isn't possible to enable only the start or the stop.*

## Other
I've fixed the warnings that IntelliJ gave me when loading the project. I'm not sure if I broke anything, but I highly doubt.

WarperSan committed:
Added gamerule for default rename

WarperSan committed:
Added sculk sensor resonance

WarperSan committed:
Fixed warnings (based of the IDE suggestions)

WarperSan committed:
Removed undesired game event

WarperSan committed:
Added sculk tests

WarperSan committed:
Cleaned up file (a little)

WarperSan committed:
Revamped the gamerule 'shouldDefaultRenameLoaders' to only affect the visibility of the name

scriptcoded commented:
Hey! Sorry, I completely missed this! I'm completely overloaded at the moment and don't have a lot of time to maintain this mod with all the breaking changes in the modding APIs... But I'll do my best to tackle this as soon as I get the time :) Sounds like great additions!

WarperSan committed:
Sync with upstream (#1)

* feat: Upgrade to Minecraft 1.21.11

* fix: build.gradle

* feat: Update to Minecraft 26.1

* fix: avoid unloading already force-loaded chunks (#51)

* chore: add TicketType CUSTOM_TICKETTYPE_FORCED

* chore: exchange setChunkForced with adding Custom Chunk Ticket

* feat: Update to MC 26.1.1 (#53)

* feat: Update to MC 26.1.2 (#54)

* fix: reduce log spam (#55)

by changing re-registration messages to debug + improved create and remove messages with more info

* feat: update to mc26.2 (#56)

* Removed unused `DispenserBlockMixin.LOGGER`

* Inlined `particleInterval`

* Fixed errors

* Removed unused imports

* Fixed gamerule

* Added back resonance

* Simplified methods

* Removed unnecessary `public` keyword

* Improved logs

* Moved game rules to `ScriptsChunkLoadersGameRules`

* Fixed small warnings

* Fixed types, logs and suppressed warning

* Added comments to tests

* Force Java 25 on toolchain

---------

Co-authored-by: Malcolm Nihlén <malcolm.nihlen@gmail.com>
Co-authored-by: Henning Raudzis <121398546+Craemon@users.noreply.github.com>

WarperSan committed:
Simplified resonance emission

WarperSan committed:
Added gamerule `emit_vibration_on_loader_change`

WarperSan committed:
Fixed typo

WarperSan committed:
Fixed name of tests

WarperSan committed:
Merge branch 'main' into main
